src/php7/php_solr_document.c

```diff
@@ -829,6 +829,51 @@ PHP_METHOD(SolrDocument, serialize)
 }
 /* }}} */
 
+/* {{{ proto array SolrDocument::__serialize(void)
+   New serializer. Returns an array with xml=>self::serialize()  */
+#if PHP_VERSION_ID >= 80100
+PHP_METHOD(SolrDocument, __serialize)
+{
+	zend_object *object = Z_OBJ_P(getThis());
+	zval serialized;
+	zend_call_method_with_0_params(object, object->ce, NULL, "serialize", &serialized);
+	if (Z_TYPE(serialized) != IS_STRING) {
+		solr_throw_exception(solr_ce_SolrIllegalArgumentException, "Internal serialize call failed", SOLR_ERROR_4000, SOLR_FILE_LINE_FUNC);
+		RETURN_THROWS();
+	}
+	array_init(return_value);
+	add_assoc_stringl(return_value, "xml", Z_STRVAL(serialized), Z_STRLEN(serialized));
```

Comment:
This could be simplified:
```suggestion
	add_assoc_str(return_value, "xml", serialized);
```

Reply:
Nice. Since `serialized` is a `zval` I used `add_assoc_zval`